### PR TITLE
feature: list models

### PR DESCRIPTION
**Pull request by ChuyangDeng**
*Issue #, if available:*

*Description of changes:*
Handle GET /models request. List the servables running on model server,

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://3r4crpxqvk.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=2b65004c-3fd9-46ed-86d1-81605687919a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: container/sagemaker/python_service.py [hunk 1]
```diff
@@ -157,6 +159,24 @@ def on_post(self, req, res):
                 'error': str(e)
             }).encode('utf-8')
 
+    def _read_model_config(self):
+        models = []
+        name_key = '    name: '
```
Comment (knakad) [inline container/sagemaker/python_service.py]: The whitespace feels brittle to me. Is there a more maintainable way to do this? Perhaps a regex search?

File: container/sagemaker/python_service.py [hunk 2]
```diff
@@ -157,6 +159,24 @@ def on_post(self, req, res):
...
+        uri_key = '    base_path: '
+        with open(MODEL_CONFIG_FILE_PATH, 'r') as f:
+            line = f.readline()
+            while line:
+                if line.startswith(name_key):
+                    model_name = line[len(name_key) + 1:len(line) - 3]
```
Comment (knakad) [inline container/sagemaker/python_service.py]: Same comment as above. I'm sure it's functional, but it triggers a lot of red flags in my mind.

File: test/integration/local/test_dynamic_endpoint.py [hunk 3]
```diff
@@ -77,6 +77,11 @@ def make_invocation_request(data, model_name, content_type='application/json'):
     return json.loads(response.content.decode('utf-8'))
 
 
+def make_list_model_request():
+    response = requests.get(MODELS_URL)
+    return json.loads(response.content.decode('utf-8'))
```
Comment (knakad) [inline test/integration/local/test_dynamic_endpoint.py]: [Minor] Is there a constant in Python that matches "utf-8"

File: container/sagemaker/python_service.py [hunk 4]
```diff
@@ -157,6 +159,24 @@ def on_post(self, req, res):
...
+                    line = f.readline()
+                    uri = line[len(uri_key) + 1:len(line) - 3]
```
Comment (knakad) [inline container/sagemaker/python_service.py]: Same as above.

File: container/sagemaker/python_service.py (hunk 1 above)
Comment (ChuyangDeng) [reply container/sagemaker/python_service.py]: I replaced this with regex search here as well as below.

File: container/sagemaker/python_service.py (hunk 2 above)
Comment (ChuyangDeng) [reply container/sagemaker/python_service.py]: Same here, used regex search to read values from model-config.cfg

File: test/integration/local/test_dynamic_endpoint.py (hunk 3 above)
Comment (ChuyangDeng) [reply test/integration/local/test_dynamic_endpoint.py]: I replaced it with `encodings.utf_8.getregentry().name`.

File: container/sagemaker/python_service.py (hunk 4 above)
Comment (ChuyangDeng) [reply container/sagemaker/python_service.py]: Done.

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://3r4crpxqvk.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=13fbb7bc-0564-428a-b842-4a480b7c2a93%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: container/sagemaker/python_service.py [hunk 5]
```diff
@@ -155,7 +159,25 @@ def on_post(self, req, res):
             res.status = falcon.HTTP_507
             res.body = json.dumps({
                 'error': str(e)
-            }).encode('utf-8')
+            }).encode(encodings.utf_8.getregentry().name)  # pylint: disable=E1101
+
+    def _read_model_config(self):
+        models = []
+        name_key = re.compile(r'( *)name:(.*)')
```
Comment (knakad) [inline container/sagemaker/python_service.py]: Awesome! I think this is much cleaner =)

[Minor] If we think the class can have tabs, we should do
```
re.compile(r'([ \t]*)name:(.*)')
```

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://3r4crpxqvk.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=be09b2e4-945a-437c-b1cc-76cc59b04e01%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**